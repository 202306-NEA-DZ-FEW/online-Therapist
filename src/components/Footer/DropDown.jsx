import Link from "next/link";
import { useTranslation } from "next-i18next";
import { useState } from "react";

const DropDown = () => {
    const [open, setOpen] = useState(false);
    const { t } = useTranslation("common");

    return (
        <div className='relative'>
            <a
                className='hover:underline cursor-pointer mb-3'
                onClick={() => setOpen(!open)}
            >
                {t("footer.about")}
            </a>
            <div className={`${open ? "block" : "hidden"} absolute`}>
                <div className='ml-4 '>
                    <Link
                        href='/about-us/'
                        className='mr-4 hover:underline md:mr-6 '
                    >
                        {t("footer.aboutUs")}
                    </Link>
                </div>
                <div className='ml-4 text-dark'>
                    <Link
                        href='/team/'
                        className='mr-4 hover:underline md:mr-6 '
                    >
                        {t("footer.team")}
                    </Link>
                </div>
            </div>
        </div>
    );
};

export default DropDown;
