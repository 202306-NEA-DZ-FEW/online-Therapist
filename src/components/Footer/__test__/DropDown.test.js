import renderer from "react-test-renderer";
import { useRouter } from "next/router";
import DropDown from "../DropDown";

// Mock the useRouter function
jest.mock("next/router", () => ({
    useRouter: jest.fn(),
}));

it("renders correctly", () => {
    // Configure the mock useRouter function to return a locale
    useRouter.mockImplementation(() => ({
        locale: "en",
    }));

    const tree = renderer.create(<DropDown />).toJSON();
    expect(tree).toMatchSnapshot();
});
