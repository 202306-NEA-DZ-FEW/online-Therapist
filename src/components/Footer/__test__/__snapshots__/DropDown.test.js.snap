// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="relative "
>
  <a
    className="mr-4 hover:underline md:mr-6 cursor-pointer p-8 mb-3  "
    onClick={[Function]}
  >
    footer.about
  </a>
  <div
    className="hidden absolute"
  >
    <div
      className="ml-6 "
    >
      <a
        className="mr-4 hover:underline md:mr-6 "
        href="#"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        footer.aboutUs
      </a>
    </div>
    <div
      className="ml-6 text-dark"
    >
      <a
        className="mr-4 hover:underline md:mr-6 "
        href="#"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        footer.team
      </a>
    </div>
  </div>
</div>
`;
